plugins/modules/lxd_storage_pool_info.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) 2025, Sean McAvoy (@smcavoy) <seanmcavoy@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: lxd_storage_pool_info
+short_description: Retrieve information about LXD storage pools
+version_added: 12.1.0
+description:
+  - Retrieve information about LXD storage pools.
+  - This module returns details about all storage pools or a specific storage pool.
+author: "Sean McAvoy (@smcavoy)"
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.info_module
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: none
+options:
+    name:
+        description:
+          - Name of a specific storage pool to retrieve information about.
+          - If not specified, information about all storage pools will be returned.
+        type: str
+    type:
+        description:
+          - Filter storage pools by driver/type (for example V(dir), V(zfs), V(btrfs), V(lvm), or V(ceph)).
+          - If not specified, all storage pool types will be returned.
+          - Can be a single type or a list of types.
+        type: list
+        elements: str
```

Comment:
OK, it seems it does make sense to have multiple choices. Is the list of types fixed? If it is we can specify them in the `choices` field.